File: ABC233/D.pl
use strict;
use warnings;

sub gets {
    chomp(my $s = <>);
    split / /,$s;
}

my ($n, $k) = gets;
my @a = gets;

my @b = (0);
for my $i (0..$n-1) {
    $b[$i+1] = $b[$i] + $a[$i];
}

my %h;
$h{$_}++ foreach(@b);

my $ans = 0;
foreach my $l (0..$n-1) {
    $h{$b[$l]}--;
    $ans += $h{$b[$l]+$k}
}
print "$ans\n";

File: ABC233/D.py
import collections

n, k = map(int, input().split())
a = list(map(int, input().split()))

b = [0] * (n+1)
for i in range(n):
    b[i+1] = b[i] + a[i]

h = collections.Counter(b)
ans = 0
for l in range(n):
    h[b[l]] -= 1
    ans += h[b[l]+k]

print(ans)